schemas/pci/pci-bus.yaml

```diff
@@ -22,139 +23,202 @@ description: |
-                - 0x03000000
-                - 0x42000000
-                - 0x43000000
-
-  "#address-cells":
-    const: 3
-
-  "#size-cells":
-    const: 2
-
-  device_type:
-    const: pci
-
-  bus-range:
-    $ref: /schemas/types.yaml#/definitions/uint32-array
-    minItems: 2
-    maxItems: 2
-    items:
-      maximum: 255
-
-  "#interrupt-cells":
-    const: 1
-
-  interrupt-map: true
-#    minItems: 1
-#    maxItems: 88    # 22 IDSEL x 4 IRQs
-#    items:
-#      minItems: 6   # 3 addr cells, 1 PCI IRQ cell, 1 phandle, 1+ parent addr and IRQ cells
-#      maxItems: 16
-
-  interrupt-map-mask:
-    items:
-      - description: PCI high address cell
-        minimum: 0
-        maximum: 0xf800
-      - description: PCI mid address cell
-        const: 0
-      - description: PCI low address cell
-        const: 0
-      - description: PCI IRQ cell
-        minimum: 0
-        maximum: 7
-
-  linux,pci-domain:
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-  max-link-speed:
-    $ref: /schemas/types.yaml#/definitions/uint32
-    enum: [ 1, 2, 3, 4 ]
-
-  num-lanes:
-    description: The number of PCIe lanes
-    $ref: /schemas/types.yaml#/definitions/uint32
-    enum: [ 1, 2, 4, 8, 16, 32 ]
-
-  reset-gpios:
-    description: GPIO controlled connection to PERST# signal
-    maxItems: 1
-
-  supports-clkreq:
-    type: boolean
-
-  aspm-no-l0s:
-    description: Disables ASPM L0s capability
-    type: boolean
-
-  vendor-id:
-    description: The PCI vendor ID
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-  device-id:
-    description: The PCI device ID
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-patternProperties:
-  "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$":
-    type: object
     properties:
+      $nodename:
+        pattern: "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$"
+
       compatible:
         contains:
           pattern: "^(pci[0-9a-f]{3,4},[0-9a-f]{1,4}|pciclass,[0-9a-f]{4,6})$"
+
+      device_type:
+        const: pci
+
       reg:
         items:
           minItems: 5
           maxItems: 5
         minItems: 1
         maxItems: 6   # Up to 6 BARs
+
+      vendor-id:
+        description: The PCI vendor ID
+        $ref: /schemas/types.yaml#/definitions/uint32
+
+      device-id:
+        description: The PCI device ID
+        $ref: /schemas/types.yaml#/definitions/uint32
+
     required:
+      - device_type
       - reg
 
-required:
-  - device_type
-  - ranges
-  - "#address-cells"
-  - "#size-cells"
+
+  pci-bridge:
+    title: PCI-to-PCI bridge function
+
+    description: |
+      A hardware device that is, from an electrical standpoint, a single
+      PCI function on one PCI bus (the "parent" bus) and the bus controller
+      of a secondary PCI bus (the "child" bus).
+
+    allOf:
+      - $ref: "#/$defs/pci-child-bus"
+      - $ref: "#/$defs/pci-function"
+      - properties:
+          compatible:
+            contains:
+              const: "pciclass,0604"
```

Comment:
This can certainly be a regular 'device' schema.

Reply:
Any reference how to correct define it?

Reply:
I mean just like 99% of typical schemas that are a single node and are applied based on compatible.